### ISSUE TITLE
RSA private key cannot be defined

### ISSUE THREAD

**Issue by zimmski** [mrjones/oauth#27]
I need to define a precomputed private key for the oauth connection for Jira as described here http://blog.michael.kuron-germany.de/2013/04/two-legged-oauth-between-php-and-jira/

I do not see how to do this. Can you give me an example on how to do this?

**Comment by zimmski** [comment]
Ok I got it, maybe you can add an example:

```go
block, _ := pem.Decode([]byte(keyRSAPrivate))
privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
if err != nil {
	...
}
```

And then for Jira use the following configuration

```go
c := oauth.NewRSAConsumer(
	keyConsumer,
	privateKey,
	oauth.ServiceProvider{
		RequestTokenUrl:   jiraURL + "/plugins/servlet/oauth/request-token",
		AuthorizeTokenUrl: jiraURL + "/plugins/servlet/oauth/authorize",
		AccessTokenUrl:    jiraURL + "/plugins/servlet/oauth/access-token",
		HttpMethod:        "POST",
	},
)
```

Since we use a self signed certificate I also just ignore the verification:
```go
c.HttpClient = &http.Client{
	Transport: &http.Transport{
		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
	},
}
```

**Comment by mrjones** [comment]
Hey cool, thanks for figuring this out.  I actually haven't used the RSASHA1 signatures before (someone else added them) so this is neat to see.

I'd love to use this in an example ... only, after a brief search, I couldn't seem to find many (or any really) OAuth service providers that use RSASHA1 signatures.  Google has a few that seem to work this way, but they claim to be shutting down all the OAuth 1.0 endpoints soon, so it would quickly become obsolete.

Do you have an example site in mind that we could use as the basis for an example?

**Comment by zimmski** [comment]
I think that an example for an imaginable Jira instance is just fine since *all* Atlassian products only support Oauth 1. Here is some documentation on this https://bitbucket.org/atlassian_tutorial/atlassian-oauth-examples I think it is OK to use the code from my comment before and let people just set the jiraURL variable to the URL of their instance e.g. "https://jira.atlassian.com" for the official public Atlassian Jira instance. You can basically pick any Atlassian service or any project which is using something from Atlassian but  the problem is that you need access to the administration to add an "application link" to register the public key.

**Comment by mrjones** [comment]
Ok, I created a jira example:

https://github.com/mrjones/oauth/blob/master/examples/jira/jira.go

I couldn't easily find a Jira project where I could be an administrator to add the Application Links you mentioned, so for now this will just have to be un-tested.  Which means it's almost certainly broken in some way :)

If you have a Jira instance that you can access, feel free to test it out, and let me know if it works.  If it's broken, I can try to fix it or I'd be happy to accept a pull request.

Thanks again for your help!

- Matt

**Comment by zimmski** [comment]
I fixed the REST API URL with #28 but other than that it works fine with our Jira instance.
